skypy/pipeline/driver.py

```diff
@@ -102,20 +110,23 @@ def isfunction(f):
                 dag.add_node(job)
 
         # Add edges for all requirements and dependencies
+        def deps(settings):
+            deps = settings.get('depends', [])
+            args = settings.get('args', [])
+            for k, v in _items(args):
+                if isinstance(v, str) and v[0] != '~':
+                    deps.append(v)
+            return deps
```

Comment:
```suggestion
        def deps(settings, dag):
            args = settings.get('args', [])
            deps = settings.get('depends', [])
            deps += [v for _, v in _items(args) if v in dag]
            return deps
```
What do you think about this alternative? Instead of requiring an escape character for string literals, just assume that if a string is in the graph of jobs then it is a dependency, otherwise assume it is a string argument. The config files would probably be cleaner but it limits the choice of variable names so that they don't clash with any string arguments. Not sure which has preferable behaviour if the user typos a variable name etc.